source/server/process_context_impl.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include "envoy/server/process_context.h"
+
+namespace Envoy {
+
+class ProcessContextImpl : public ProcessContext {
+public:
+  ProcessContextImpl(ProcessObject* process_object) : process_object_(process_object) {}
```

Comment:
Should we be using `ProcessObject&` here? I.e. is this always non-null? Same for the rest of the `ProcessContext` interface.